Reflection/reflection.h

```diff
@@ -48,28 +73,29 @@ struct ReflectorImpl {
     }
 
     template <typename T>
-    typename std::enable_if<std::is_same<SuperType<T>, CurrentSuper>::value, std::string>::type
-    StructInheritanceString() {
-      return "";
-    }
-
-    template <typename T>
-    typename std::enable_if<!std::is_same<SuperType<T>, CurrentSuper>::value, std::string>::type
-    StructInheritanceString() {
-      return std::string(" : ") + SuperType<T>::template CURRENT_REFLECTION_HELPER<SuperType<T>>::name();
-    }
-
-    template <typename T>
     typename std::enable_if<std::is_base_of<CurrentSuper, T>::value, std::unique_ptr<ReflectedTypeImpl>>::type
     operator()(TypeSelector<T>) {
       auto s = make_unique<ReflectedType_Struct>();
       s->name = T::template CURRENT_REFLECTION_HELPER<T>::name();
-      s->super_name = StructInheritanceString<T>();
+      s->super_name = SuperNameIfNotCurrentSuper<T>();
```

Comment:
Actually, we'll need to return not just `struct Foo {` but `struct Foo : CurrentSuper {` in C++ reflected types.